server/sonar-server/src/main/java/org/sonar/server/settings/ws/SetAction.java

```diff
@@ -203,6 +301,24 @@ private static SetRequest toWsRequest(Request request) {
     return setRequest;
   }
 
+  private static List<Map<String, String>> readFieldValues(Request request) {
+    String key = request.mandatoryParam(PARAM_KEY);
+
+    return request.multiParam(PARAM_FIELD_VALUES).stream()
+      .map(json -> readOneFieldValues(json, key))
+      .collect(Collectors.toList());
+  }
+
+  private static Map<String, String> readOneFieldValues(String json, String key) {
+    Type type = new TypeToken<Map<String, String>>() {
+    }.getType();
+    try {
+      return (Map<String, String>) GsonHelper.create().fromJson(json, type);
+    } catch (JsonParseException e) {
+      throw new BadRequestException(String.format("Ill-formatted JSON '%s' for setting '%s'", json, key));
```

Comment:
As reported by my first comment, the error message is strange. Wha means "Ill-formated" ?